dev/devicelab/lib/tasks/perf_tests.dart

```diff
@@ -1339,22 +1348,58 @@ class WebCompileTest {
     });
   }
 
-  /// Obtains the size and gzipped size of a file given by [fileName].
-  static Future<Map<String, int>> getSize(String fileName, {required String metric}) async {
+  /// Obtains the size and gzipped size of both [dartBundleFile] and [buildDir].
+  static Future<Map<String, int>> getSize({
+    /// Mapping of metric key name to file system path for directories to measure
+    Map<String, String> directories = const <String, String>{},
+    /// Mapping of metric key name to file system path for files to measure
+    Map<String, String> files = const <String, String>{},
+    required String metric,
+  }) async {
+    const String kGzipCompressionLevel = '-9';
     final Map<String, int> sizeMetrics = <String, int>{};
 
-    final ProcessResult result = await Process.run('du', <String>['-k', fileName]);
-    sizeMetrics['${metric}_dart2js_size'] = _parseDu(result.stdout as String);
+    final Directory tempDir = Directory.systemTemp.createTempSync('perf_tests_gzips');
+    for (final MapEntry<String, String> entry in files.entries) {
+      final String key = entry.key;
+      final String filePath = entry.value;
+      sizeMetrics['${metric}_${key}_uncompressed_bytes'] = File(filePath).lengthSync();
 
-    await Process.run('gzip',<String>['-k', '9', fileName]);
-    final ProcessResult resultGzip = await Process.run('du', <String>['-k', '$fileName.gz']);
-    sizeMetrics['${metric}_dart2js_size_gzip'] = _parseDu(resultGzip.stdout as String);
+      await Process.run('gzip',<String>['--keep', kGzipCompressionLevel, filePath]);
+      // gzip does not provide a CLI option to specify an output file, so
+      // instead just move the output file to the temp dir
+      final File compressedFile = File('$filePath.gz')
+          .renameSync(path.join(tempDir.absolute.path, '$key.gz'));
+      sizeMetrics['${metric}_${key}_compressed_bytes'] = compressedFile.lengthSync();
 
-    return sizeMetrics;
-  }
+    }
 
-  static int _parseDu(String source) {
-    return int.parse(source.split(RegExp(r'\s+')).first.trim());
+    for (final MapEntry<String, String> entry in directories.entries) {
+      final String key = entry.key;
+      final String dirPath = entry.value;
+      final Directory dir = Directory(dirPath);
+      sizeMetrics['${metric}_${key}_uncompressed_bytes'] = dir
+          .listSync(recursive: true)
+          .whereType<File>()
+          .fold<int>(
+            0,
+            (int totalSize, File file) => totalSize + file.lengthSync(),
+          );
+
+      // get size of compressed build directory
```

Comment:
OK, crazy idea that might simplify this and help reuse code: hear me out...

`tar` (without gzip) the build directory, and get the du for that file -> uncompressed directory size

then `gzip` the tar that you created above and measure the size -> compressed directory size

That way you don't have to manually iterate through the filesystem with dart:io

Reply:
After the first line I was getting ready to say no...but this is a really good idea.